### PR TITLE
[RFC/RFT] bind: update to 9.14.2

### PR DESCRIPTION
**Pull request by dengqf6**
Maintainer: @nmeyerhans 
Compile tested: ramips, ath79, mvebu, octeon, brcm2708, gemini
Run tested:  ramips, mvebu

Description:
Update bind to 9.14.2
BIND now requires POSIX thread to build
Add filter-AAAA plugin
Remove unrecognized options
Remove patch that no longer needed
- 002-autoconf-ar-fix.patch

Select libatomic for MIPS

### CODE REVIEW

**Comment by neheb** [comment]
I believe it's needed for ARM as well if memory serves me correctly.

**Comment by dengqf6** [comment]
@neheb Just compile tested on mvebu. Doesn't need libatomic.

**Comment by dengqf6** [comment]
I marked this as RFC because I've encountered this error:
```
checking for yield instruction support... configure: error: no, try adding -march=native or -march=armv7-a to CFLAGS (see PLATFORMS.md for more information)
```
According to `PLATFORMS.md`:

### ARM

If the compilation ends with following error:

```
Error: selected processor does not support `yield' in ARM mode
```

You will need to set `-march` compiler option to `native`, so the compiler
recognizes `yield` assembler instruction.  The proper way to set `-march=native`
would be to put it into `CFLAGS`, e.g. run `./configure` like this:
`CFLAGS="-march=native -Os -g" ./configure` plus your usual options.

If that doesn't work, you can enforce the minimum CPU and FPU (taken from Debian
armhf documentation):

* The lowest worthwhile CPU implementation is Armv7-A, therefore the recommended
  build option is `-march=armv7-a`.
---
So it doesn't support ARMv6 or lower anymore?

**Comment by diizzyy** [comment]
Can you share the full build log on ARM and it is running head or 18.X?

**Comment by dengqf6** [comment]
OpenWrt master, of course.
Builds and runs on mvebu.

**Comment by diizzyy** [comment]
Just disable it for < ARMv7, hacking around it will just be another patch we need to maintain and most have moved on from ARMv5 and ARMv6 by now.

**Comment by ja-pa** [comment]
@dengqf6 You should bump version to 9.14.1 since there are multiple security issues affecting 9.14.0 version

 * CVE-2018-5743 - Limiting simultaneous TCP clients is ineffective https://kb.isc.org/docs/cve-2018-5743

 * CVE-2019-6467 - An error in the nxdomain redirect feature can cause BIND to exit with an INSIST assertion failure in query.c https://kb.isc.org/docs/cve-2019-6467

 * CVE-2019-6468 - BIND Supported Preview Edition can exit with an assertion failure if nxdomain-redirect is used https://kb.isc.org/docs/cve-2019-6468

**Comment by diizzyy** [comment]
Preferably you shouldn't run bind at all because of security concerns *ducks* ;-)

**Comment by dengqf6** [comment]
Updated

File: net/bind/Makefile [hunk 1]
```diff
@@ -122,20 +127,14 @@ export BUILD_CC="$(TARGET_CC)"
...
 	--disable-linux-caps \
 	--with-openssl="$(STAGING_DIR)/usr" \
 	--with-libtool \
 	--without-lmdb \
 	--enable-epoll=yes \
-	--with-gost=no \
 	--with-gssapi=no \
-	--with-ecdsa=$(if $(CONFIG_OPENSSL_WITH_EC),yes,no) \
-	--with-eddsa=$(if $(CONFIG_BIND_ENABLE_EDDSA),yes,no) \
 	--with-readline=no \
```
Comment (neheb) [inline net/bind/Makefile]: Can be changed to --without.

File: net/bind/Makefile [hunk 2]
```diff
@@ -56,7 +55,7 @@ endef
 define Package/bind-libs
   SECTION:=libs
   CATEGORY:=Libraries
-  DEPENDS:=+libopenssl +zlib
+  DEPENDS:=+libopenssl +zlib +libpthread +(mips||mipsel):libatomic
```
Comment (neheb) [inline net/bind/Makefile]: That libatomic thing is fascinating... I wonder if it also applies to mips64.

File: net/bind/Makefile [hunk 3]
```diff
@@ -122,20 +127,14 @@ export BUILD_CC="$(TARGET_CC)"
...
-	--enable-ipv6=$(if $(CONFIG_IPV6),yes,no) \
-	--with-randomdev="/dev/urandom" \
-	--disable-threads \
 	--disable-linux-caps \
 	--with-openssl="$(STAGING_DIR)/usr" \
 	--with-libtool \
 	--without-lmdb \
 	--enable-epoll=yes \
-	--with-gost=no \
 	--with-gssapi=no \
```
Comment (neheb) [inline net/bind/Makefile]: Can be changed to --without.

File: net/bind/Makefile [hunk 4]
```diff
@@ -122,20 +127,14 @@ export BUILD_CC="$(TARGET_CC)"
...
 
 CONFIGURE_ARGS += \
-	--enable-ipv6=$(if $(CONFIG_IPV6),yes,no) \
-	--with-randomdev="/dev/urandom" \
-	--disable-threads \
 	--disable-linux-caps \
 	--with-openssl="$(STAGING_DIR)/usr" \
 	--with-libtool \
 	--without-lmdb \
 	--enable-epoll=yes \
```
Comment (neheb) [inline net/bind/Makefile]: the =yes can be removed I think.

**Comment by dengqf6** [comment]
Compile tested on octeon (mips64), doesn't need libatomic

**Comment by dengqf6** [comment]
Compile tested on brcm2708 (ARMv6 with yield instruction)

File: net/bind/Makefile (hunk 2 above)
Comment (dengqf6) [reply net/bind/Makefile]: Seems gemini needs libatomic as well..

**Comment by dengqf6** [comment]
@nmeyerhans ping

**Comment by nmeyerhans** [comment]
@dengqf6 Thanks for this work, it looks good in general. I'd like a little more time to read it in depth and test.

File: net/bind/Makefile [hunk 5]
```diff
@@ -122,41 +127,30 @@ export BUILD_CC="$(TARGET_CC)"
...
-	--disable-threads \
 	--disable-linux-caps \
 	--with-openssl="$(STAGING_DIR)/usr" \
 	--with-libtool \
 	--without-lmdb \
-	--enable-epoll=yes \
-	--with-gost=no \
-	--with-gssapi=no \
-	--with-ecdsa=$(if $(CONFIG_OPENSSL_WITH_EC),yes,no) \
-	--with-eddsa=$(if $(CONFIG_BIND_ENABLE_EDDSA),yes,no) \
```
Comment (nmeyerhans) [inline net/bind/Makefile]: Where did these options go? Did bind drop support for these features, or are they enabled unconditionally now, or what? I haven't been able to find anything in the release notes to clarify this, but I'd like to understand the change before we merge this.

Comment (dengqf6) [reply net/bind/Makefile]: ECDSA support in OpenSSL is mandatory. Other options are nowhere to be found (Do not get recognized by `./configure`)

**Comment by dengqf6** [comment]
Updated to 9.14.2

File: net/bind/Makefile (hunk 5 above)
Comment (nmeyerhans) [reply net/bind/Makefile]: Looks like the check for EDDSA and others are implicit, and can't be disabled. So this is fine.

File: net/bind/Makefile [hunk 6]
```diff
@@ -56,7 +55,7 @@ endef
 define Package/bind-libs
   SECTION:=libs
   CATEGORY:=Libraries
-  DEPENDS:=+libopenssl +zlib
+  DEPENDS:=+libopenssl +zlib +libpthread +(mips||mipsel||TARGET_gemini):libatomic
```
Comment (neheb) [inline net/bind/Makefile]: I don't like this. How about we just add libatomic for all targets?

Comment (nmeyerhans) [reply net/bind/Makefile]: What don't you like about it? We don't want unnecessary dependencies. If libatomic is only needed for certain targets, it should only be a dependency on those targets.

Comment (neheb) [reply net/bind/Makefile]: Sure. The main concern is whether or not this is a moving target. libatomic itself seems to be around 6kb. Not too big.

Comment (nmeyerhans) [reply net/bind/Makefile]: OK, I've looked through the rest of the packages repository, and haven't found any others that try to conditionally enable the libatomic dependency. They all add it globally if they use it at all. So let's just follow that precedent.

Comment (dengqf6) [reply net/bind/Makefile]: I see

**Comment by dengqf6** [comment]
Is this okay to be merged now?

**Pull request merged**